RMET-2802 InAppBrowser Plugin - Multiple minor fixes from the community

alexgerardojacinto opened the pull request:
## Description
<!--- Describe your changes in detail -->

- This PR contains 3 fixes that came from the community, from the following PRs: https://github.com/OutSystems/cordova-plugin-inappbrowser/pulls?q=is%3Apr+author%3Aleonardo-fernandes.
- The fixes include:
    - Not loosing the `callbackContext` when opening a `SYSTEM_URL`
    - Fixing beforeLoad not being called in every request.
    - Fixing a memory leak - webview was being left in memory unnecessarily. 
- A more thorough explanation for each fix is available on each PR.
- The PR also includes some refactors regarding ternary operators.

## Context
<!--- Why is this change required? What problem does it solve? -->
<!--- Place the link to the issue here -->
References: https://outsystemsrd.atlassian.net/browse/RMET-2802

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [x] Fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [ ] iOS
- [x] JavaScript

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->

- Tested on both Android (12 and 14) and iOS (17), with the InAppBrowser Sample App (tested every feature in the app).

- MABS 9 Android build: https://intranet.outsystems.net/MABS/BuildDetail.aspx?BuildId=bd50be7f72b1bfac4de1a0241117b081eb641694
- MABS 9 iOS build: https://intranet.outsystems.net/MABS/BuildDetail.aspx?BuildId=713c7bb486a8f916997d349a095f98b02c965d3d

## Screenshots (if appropriate)

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

leonardo-fernandes committed:
Fix memory leak - initially reported in https://github.com/apache/cordova-plugin-inappbrowser/issues/290

Whenever closing an InAppBrowser instance, a webview was left in memory with about:blank page.
This change fixes the issue by destroying and freeing the inAppWebView object.

leonardo-fernandes committed:
Fix beforeLoad not being called in some requests - initially reported in https://github.com/apache/cordova-plugin-inappbrowser/issues/686

The waitForBeforeload flag was preventing beforeLoad from being called on every GET request.

leonardo-fernandes committed:
Do not lose callbackContext when opening a SYSTEM url

This fixes a condition where it was not possible to open a SYSTEM url while an InAppBrowser instance is displayed. The callbackContext of the InAppBrowser instance was lost when the SYSTEM url was opened. This fixes the issue by not setting the callbackContext on SYSTEM urls.

leonardo-fernandes committed:
Fix crash when pausing/resuming application after closing InAppBrowser

leonardo-fernandes committed:
Fix _loadAfterBeforeload callback not working due to this.rootName not being defined.

leonardo-fernandes committed:
Reset beforeload variable if a new instance of InAppBrowser is opened without beforeload setting

nflsilva committed:
Merge pull request #37 from leonardo-fernandes/outsystems-fix-system-callback

Do not lose callbackContext when opening a SYSTEM url

nflsilva committed:
Merge pull request #36 from leonardo-fernandes/outsystems-fix-before-load

Fix beforeLoad not being called in some requests

nflsilva committed:
Merge pull request #35 from leonardo-fernandes/outsystems-fix-memory-leak

Fix memory leak

alexgerardojacinto committed:
chore: update changelog

References: https://outsystemsrd.atlassian.net/browse/RMET-2802

alexgerardojacinto committed:
Merge branch 'development' into fix/RMET-1290/prs-external-fixes

OS-ricardomoreirasilva commented on src/android/InAppBrowser.java:
```diff
@@ -721,6 +729,8 @@ public String showWebPage(final String url, HashMap<String, String> features) {
             }
             if (features.get(BEFORELOAD) != null) {
                 beforeload = features.get(BEFORELOAD);
+            } else {
```
ternary operator?

alexgerardojacinto replied:
Yup, done!

OS-ricardomoreirasilva commented on CHANGELOG.md:
```diff
@@ -4,7 +4,15 @@ All notable changes to this project will be documented in this file.
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
+
```
No need for these two empty lines.

alexgerardojacinto committed:
refactor: remove empty lines

alexgerardojacinto committed:
refactor: use ternary operator and remove unnecessary ones

Why: We can refactor the if-else block that assigns a value to beforeload to use a ternary operator. On the other hand, we can remove the unnecessary ternary operators used to assign a boolean value. They are redundant.

References: https://outsystemsrd.atlassian.net/browse/RMET-2802
